pkg/asset-manager-utils/contracts/test/TestAssetManager.sol

```diff
@@ -43,31 +42,22 @@ contract TestAssetManager is RewardsAssetManager {
      * @param aum - the current assets under management of this asset manager
      * @return the number of shares to mint for the pool
      */
-    function _invest(uint256 amount, uint256 aum) internal override returns (uint256) {
-        nextAUM = aum + amount;
+    function _invest(uint256 amount, uint256 aum) internal pure override returns (uint256) {
         return amount;
     }
 
     /**
      * @param amount - the amount of tokens being divested
      * @return the number of tokens to return to the vault
      */
-    function _divest(uint256 amount, uint256 aum) internal override returns (uint256) {
-        nextAUM = aum - amount;
+    function _divest(uint256 amount, uint256 aum) internal pure override returns (uint256) {
         return amount;
     }
 
     /**
      * @return the current assets under management of this asset manager
      */
     function readAUM() public view override returns (uint256) {
-        return nextAUM;
-    }
-
-    /**
-     * @dev Sets the next value to be read by `readAUM`
-     */
-    function setUnrealisedAUM(uint256 aum) external {
-        nextAUM = aum;
+        return token.balanceOf(address(this));
```

Comment:
This is fantastic, much more straightforward to understand.

If we add a `burn` function to `TestToken`, we can do the same thing for `capitalOut`.